service/history/shard/context_factory.go

```diff
@@ -76,6 +77,8 @@ type (
 		ThrottledLogger             log.ThrottledLogger
 		TimeSource                  clock.TimeSource
 		TaskCategoryRegistry        tasks.TaskCategoryRegistry
+		EventsCache                 events.Cache
+		NewEventsCacheFn            events.NewEventsCacheFn
```

Comment:
do we need fn here. it is go convention to use New in front of initializing function name.

Reply:
I just removed this function.